extension/src/Xliff/XLIFFDocument.ts

```diff
@@ -586,6 +589,16 @@ export class TransUnit implements TransUnitInterface {
     if (_maxwidthText) {
       _maxwidth = Number.parseInt(_maxwidthText);
     }
+    let _approved: boolean | undefined = undefined;
+    const _approvedText = transUnit.getAttributeNode("approved")?.value;
+    if (_approvedText) {
+      _approved =
+        _approvedText === null ||
+        _approvedText === undefined ||
+        _approvedText.toLowerCase() === "no"
+          ? false
+          : true;
```

Comment:
Haven't tested this but couldn't it just be this?
```suggestion
      _approved = _approvedText?.toLowerCase() === "yes";
```

Reply:
We need the "no" part as well... and it can be omitted.

Actually, I just copy&pasted your old code for the "translate" attribute, without that much thinking 😜

Reply:
hmmm... I need to double check this, don't even understand why tests are working...

Reply:
I might be missing something but this assignment resides within a check for truthy so it should be either "no", "yes" or true?

Reply:
you where right, I looked to fast at your comment and didn't check the context of your suggestion :)

And since it was guarded against null/undefined I could even remove the ? part.

But your suggestion applied perfectly to the translate attribute, so updated with that as well.